File: tests/integration/construct_component/python/component.py
# Copyright 2016-2020, KhulnaSoft Ltd.  All rights reserved.

from typing import Any, Optional

import codeinfra

class Component(codeinfra.ComponentResource):
    echo: codeinfra.Output[Any]
    childId: codeinfra.Output[str]

    def __init__(self, name: str, echo: codeinfra.Input[Any], opts: Optional[codeinfra.ResourceOptions] = None):
        props = dict()
        props["echo"] = echo
        props["childId"] = None
        props["secret"] = None
        super().__init__("testcomponent:index:Component", name, props, opts, True)

File: tests/integration/construct_component/nodejs/component.ts
// Copyright 2016-2020, KhulnaSoft Ltd.  All rights reserved.

import * as codeinfra from "@codeinfra/codeinfra";

interface ComponentArgs {
    echo: codeinfra.Input<any>;
}

export class Component extends codeinfra.ComponentResource {
    public readonly echo!: codeinfra.Output<any>;
    public readonly childId!: codeinfra.Output<codeinfra.ID>;

    constructor(name: string, args: ComponentArgs, opts?: codeinfra.ComponentResourceOptions) {
        const inputs: any = {};
        inputs["echo"] = args.echo;
        inputs["childId"] = undefined /*out*/;
        inputs["secret"] = undefined /*out*/;

        super("testcomponent:index:Component", name, inputs, opts, true);
    }
}

